### PR TITLE
Support ViT models in HF test

### PR DESCRIPTION
**Pull request by makslevental**
Support Vision Transformer models, fixes https://github.com/pytorch/PiPPy/issues/244

cc @desertfire

### CODE REVIEW

File: test/hf_test.py [hunk 1]
```diff
@@ -391,9 +417,8 @@ def test_case(self):
             if model_cls in [SwinForMaskedImageModeling, SwinForImageClassification, SwinModel]:
                 self.skipTest('Need to support SWIN models')
 
-            # TODO: support ViT models https://github.com/pytorch/PiPPy/issues/244
-            if model_cls in [ViTForMaskedImageModeling, ViTForImageClassification, ViTModel]:
-                self.skipTest('Need to support ViT models')
+            if model_cls in [ViTModel]:
```
Comment (pbelevich) [inline test/hf_test.py:420]: Instead of this please add `ViTModel` [here](https://github.com/pytorch/PiPPy/blob/ffa34589bc5e2981b1230d5a4b4625035eca1fb5/test/hf_test.py#L466)

File: test/hf_test.py [hunk 2]
```diff
@@ -369,6 +392,9 @@ def get_output_loss_value_spec_for_model(model_cls):
                      PLBartForSequenceClassification, PegasusForConditionalGeneration, BartForConditionalGeneration]:
         return {'loss': True, 'logits': False, 'encoder_last_hidden_state': False}
 
+    if model_cls in [ViTForMaskedImageModeling, ViTForImageClassification, ViTModel]:
+        return {'loss': True, 'logits': True}
```
Comment (pbelevich) [inline test/hf_test.py:396]: Why 'logits': True? True should be specified for the variable that is supposed to be "loss"
```suggestion
        return {'loss': True, 'logits': False}
```

**Review by pbelevich — CHANGES_REQUESTED** [review]
thanks for working on this!

File: test/hf_test.py (hunk 2 above)
Comment (makslevental) [reply test/hf_test.py:396]: sorry that's a holdover from before i figured out that the right API to get the models to return loss at all. thanks for the catch

**Review by pbelevich — APPROVED** [review]
LGTM
